options.py

```diff
@@ -204,9 +204,15 @@ def update_from_permalink(self, permalink: str):
                 raise Exception(f'unknown type: {option["type"]}')
             self.set_option(option_name, value)
 
-    def to_dict(self):
+    def to_dict(self, include_nonperma=True, exclude=[]):
         opts = self.options.copy()
         for option_name, option in OPTIONS.items():
+            if not include_nonperma and "permalink" in option and not option["permalink"]:
```

Comment:
```suggestion
            if exclude_nonperma and option.get("permalink", True) is False:
```
Of course the `exclude_nonperma` would have to be reflected at all (1) call sites